src/jit/lsraxarch.cpp

```diff
@@ -2446,75 +2480,62 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
             }
 #endif
             break;
+        }
 
 #ifdef _TARGET_X86_
         case NI_SSE42_Crc32:
         {
-            // CRC32 may operate over "byte" but on x86 only RBM_BYTE_REGS can be used as byte registers.
-            //
             // TODO - currently we use the BaseType to bring the type of the second argument
             // to the code generator. May encode the overload info in other way.
-            var_types srcType       = intrinsicTree->gtSIMDBaseType;
-            regMaskTP op2Candidates = RBM_NONE;
-            BuildUse(op1);
-            if (varTypeIsByte(srcType))
-            {
-                op2Candidates = allByteRegs();
-            }
-            RefPosition* op2Use = BuildUse(op2, op2Candidates);
+
+            assert(numArgs == 2);
             assert(isRMW);
-            setDelayFree(op2Use);
-            srcCount  = 2;
+
+            // CRC32 may operate over "byte" but on x86 only RBM_BYTE_REGS can be used as byte registers.
+            srcCount += BuildOperandUses(op1);
+            srcCount += BuildDelayFreeUses(op2, varTypeIsByte(baseType) ? allByteRegs() : RBM_NONE);
+
             buildUses = false;
             break;
         }
 #endif // _TARGET_X86_
 
         default:
+        {
             assert((intrinsicID > NI_HW_INTRINSIC_START) && (intrinsicID < NI_HW_INTRINSIC_END));
             break;
+        }
     }
 
     if (buildUses)
     {
-        if (numArgs > 3)
```

Comment:
There are currently 0 cases where we have an intrinsic with more than 3 args that would reach the register allocator.